src/map_command_handler.hpp

```diff
@@ -191,9 +191,21 @@ class map_command_handler
 		else if (help_on_unknown_) {
 			utils::string_map symbols;
 			symbols["command"] = get_cmd();
-			symbols["help_command"] = cmd_prefix_ + "help";
-			print("help", VGETTEXT("Unknown command '$command', try $help_command "
-				"for a list of available commands.", symbols));
+			std::string input = get_data(0);
+			if (cmd_prefix_ == "/"){
+				symbols["help_command"] = cmd_prefix_ + "help";
+			}
+			else{
+				symbols["help_command"] = "help";
+			}
+			if (input[0] == ':'){
+				print("help", VGETTEXT("Unknown command '$command', remove ':' or try '$help_command' "
+					"for a list of available commands.", symbols));
+			}
+			else{
+				print("help", VGETTEXT("Unknown command '$command', try '$help_command' "
```

Comment:
Something that could be left to a follow-up PR would be to convert the `'` marks to match the https://wiki.wesnoth.org/Typography_Style_Guide . There are other strings in this file that aren't involved in the current change, so I'd be happy to postpone all of these to a follow-up.

Reply:
I would agree.